lib/rules/no-unused-vars.js

```diff
@@ -204,16 +215,10 @@ module.exports = {
          */
         function hasRestSpreadSibling(variable) {
             if (config.ignoreRestSiblings) {
-                return variable.defs.some(def => {
-                    const propertyNode = def.name.parent;
-                    const patternNode = propertyNode.parent;
-
-                    return (
-                        propertyNode.type === "Property" &&
-                        patternNode.type === "ObjectPattern" &&
-                        REST_PROPERTY_TYPE.test(patternNode.properties[patternNode.properties.length - 1].type)
-                    );
-                });
+                const hasRestSiblingDef = variable.defs.some(def => hasRestSibling(def.name.parent));
```

Comment:
Could we extend these variable names to hasRestSiblingDefinition and hasRestSiblingReference? Right now these are a bit hard to distinguish from one another.

Reply:
@nzakas Thanks for the review. I fixed it. 👍